Breakout3

GBruening opened the pull request:
Addressing #3 

Added requested changes that undo some of reviewer two comments.

Changed a few of reviewer 2's comments to make them look bad. Changed from int/255 to the decimal values so the code is consistant.

Split the main subscripts and solutions. We can figure out if we want this or not.

GBruening committed:
chaning name of breakout 2 mark down

GBruening committed:
genpath is bad, and with git files it makes this take to long. Just use the two addpaths and its also better practice

GBruening committed:
Changed some things with breakout 2 to make the peach not look good. If it looks good then theres no reason for breakout 3

GBruening committed:
Added 3 commnets for reviewer 3. 2 of them to revert changes, 1 to change one last thing. Also made the code more consistnat across color choices. We should stick to either decimals or 255 values (though i know it gets converted by the /255)

GBruening committed:
Added files that contain only the solutions

GBruening committed:
Added hit to the breakout3 markdown for the reverting changes. We probalby need to discuss how to do this reverting changes

GBruening committed:
Cleaning up the subscript files

GBruening committed:
Changed the hint a bit

alcantarar commented:
Thanks for working on this! I'm gonna hold off on a review until after tomorrow's meeting.

alcantarar commented on breakout_3_comments.md:
```diff
@@ -0,0 +1,16 @@
+# Reviewer Comments
```
storyline for breakout 3 is:
- after making revisions, paper got rejected. 😢 
- decide to submit to another journal (Journal of Biomechanics - really the next best option besides Nature 😉 ).
- undo the bad reviewer's comments and make some final touches before re-submission.

alcantarar commented on breakout_3_comments.md:
```diff
@@ -0,0 +1,16 @@
+- Revert the changes done to the leafs color to its original color. **_Refer to script 2_**
+- The text for 'Virtual 2020' is to large and overlapping the line. Reduce the text size to 16. **_Refer to script 5_**
+
+** Hint: You can find the commit that the changes were made and checkout to that commit, then re-commit this for two of these comments.
```
are you going to provide any code examples on how to actually do this in git? This handout should walk the user through the process of reverting changes, not just telling them to do so. It's a tutorial, not a quiz.

GBruening replied:
We used to have that in the gitcheat sheet, and this should be addressed in your presentation. 

Breakout 3 is not the place to address this. This should be told to them much more in advance.

GBruening replied:
I thought this was addressed during breakout 1, it's not. So i guess we do it here.

alcantarar replied:
> I hate the solutions in the script files.

🙄 

> If you want to put them there I'll let you. But if you don't want them copying and pasting like you said in another comment, then they shouldn't be there. The changes are so simple, if they can't figure it out they won't be able to figure out git.

Yep, no solution files. It was simpler before and allows for easy reference because again, the point of the tutorial isn't how to debug or see if they can figure it out. We need this tutorial to be as user-friendly as possible.

> The markdown had information on checking out to a specific branch, and I'm not sure where it went. I don't think it should be addressed here how to checkout to specific commits but should be done earlier.
>
> This checkout method is going to be a little odd. Because they 'll need to go into their branch and checkout that branch, then pull request again.

We can talk tomorrow about what this breakout room is/isn't supposed to be about.

alcantarar commented on main_script.m:
```diff
@@ -2,7 +2,9 @@
 %   Each of the scripts below has a bug that needs to be fixed!
 %
-addpath(genpath('..'))
+addpath(strcat(pwd,'\subscripts'))
```
I think `pwd` just gets current folder. Make sure it's clear that users need the working directory to be the current folder in matlab. This is the same issue with `genpath`: someone might open matlab before the scripts, so then the current folder is wherever they left off their last matlab session. This `strcat` method is fine.

GBruening replied:
When you run a script matlab asks two things. Either change folder, or add file directory as a path. 
This code works either choice they make.

alcantarar replied:
add a comment explaining this.

alcantarar commented on breakout_3_comments.md:
```diff
@@ -0,0 +1,16 @@
+### General comments
+I thought this figure looked very good before? I'm not sure what silly reviewer wanted changes done to it, but it doesn't look good anymore!The colors are absurd! I have listed my comments on what to revert or change.
+
+### Specific comments
+- Revert the changes done to the peaches color to its original color. **_Refer to script 1_**
+- Revert the changes done to the leafs color to its original color. **_Refer to script 2_**
+- The text for 'Virtual 2020' is to large and overlapping the line. Reduce the text size to 16. **_Refer to script 5_**
```
I only see 3 changes for this breakout. What are the other group members supposed to do? Each script should have a commit to revert back to so everyone practices this. Then, you can give some an additional change to make it fancier. 

We don't want someone to just go "oh I remember what the RGB was, I'll type it in". We want them to actually revert back to a particular commit. To encourage this behavior, the intro paragraph for this handout can say something like "After months of waiting to hear back from Nature...", implying that it's been so long since you made the last changes that you don't remember and should just revert to the commit. Hopefully their commit message from before is informative! Walk them through the process I outlined for you here: https://github.com/alcantarar/ASB_Tutorial/issues/3#issuecomment-655848607

GBruening replied:
I think leaving them with only 3 changes is fine, as reverting or checking out to previous commits is a little more complicated than the other stuff they've done.

alcantarar replied:
But there are 5 people in a group and 3 things to change. What are the other group members supposed to do?

alcantarar commented on subscripts/script_1_sol.m:
```diff
@@ -0,0 +1,25 @@
+%read in data
+data = csvread('peach.csv');
+%plot data
+p = patch(data(:,1), data(:,2), [0.9882 0.1961 0.3490]); % divide by 255 to normalize value between 0 and 1
```
remove comment on line 16 because handout no longer requires division by 255.

alcantarar commented on subscripts/script_1_sol.m:
```diff
@@ -0,0 +1,25 @@
+p = patch(data(:,1), data(:,2), [0.9882 0.1961 0.3490]); % divide by 255 to normalize value between 0 and 1
+p.LineWidth = 1; % change to 1
+
+%% SOLUTION: Breakout 3
+
+%read in data
+data = csvread('peach.csv');
+%plot data
+p = patch(data(:,1), data(:,2), [0.8980 0.3176 0.1765]); % divide by 255 to normalize value between 0 and 1
+p.LineWidth = 1; % change to 1
```
comments on line 24 and 25 relate to breakouts 1 and 2. Remove from Breakout room 3 solution and update comments to match whatever you'll end up having them change.

alcantarar commented on subscripts/script_2_sol.m:
```diff
@@ -0,0 +1,26 @@
+%read in data
+data = csvread('leaf.csv');
+%plot data
+l = patch(data(:,1), data(:,2), [0.5686 0.5882 0.3765]); % divide by 255 to normalize value between 0 and 1
```
rm comment on line 17 because rgb is explicit in handout now.

alcantarar commented on subscripts/script_2_sol.m:
```diff
@@ -0,0 +1,26 @@
+l = patch(data(:,1), data(:,2), [0.5686 0.5882 0.3765]); % divide by 255 to normalize value between 0 and 1
+l.LineWidth = 1; % change to 1
+
+%% SOLUTION: Breakout 2
+
+%read in data
+data = csvread('leaf.csv');
+%plot data
+l = patch(data(:,1), data(:,2), 'green'); % divide by 255 to normalize value between 0 and 1
+l.LineWidth = 1; % change to 1
```
1. this says "breakout 2" but should be breakout 3
2. comments are not relevant for breakout 3

alcantarar commented on subscripts/script_3_sol.m:
```diff
@@ -0,0 +1,36 @@
+plot(data(:,1), data(:,2), 'k-', 'LineWidth',1); % change linewidth to 1
+plot(data(:,3), data(:,4), 'k-', 'LineWidth',1); % change linewidth to 1
+
+%% SOLUTION: Breakout 3
+
+%read in data
+data = csvread('outline.csv');
+%remove missing data 
+data(data == -999) = NaN;
+%plot data
+hold on
+plot(data(:,1), data(:,2), 'k-', 'LineWidth',1); % change linewidth to 1
+plot(data(:,3), data(:,4), 'k-', 'LineWidth',1); % change linewidth to 1
```
comments on line 34/35 not relevant to breakout 3.

alcantarar commented on subscripts/script_4_sol.m:
```diff
@@ -0,0 +1,24 @@
+%plot data
+plot(data(:,1), data(:,2), 'k-', 'LineWidth',4); % change linewidth to 4
+
+%% SOLUTION: Breakout 2
+
+%read in data
+data = csvread('ASB.csv');
+%plot data
+plot(data(:,1), data(:,2), 'k-', 'LineWidth',4); % change linewidth to 4
+
```
1. says "breakout 2" when it should read "breakout 3"
2. comments on line 23 not relevant to breakout 3.

alcantarar commented on subscripts/script_5_sol.m:
```diff
@@ -0,0 +1,24 @@
+%plot text
+text(2.5,-3.5, ['Virtual ' year], 'FontSize', 24, 'FontWeight', 'bold') % Include 'Virtual ' before 2020
+
+%% SOLUTION: Breakout 2
+
+%today's date
+year = date_text(end-3:end);
+%plot text
+text(2.5,-3.5, ['Virtual ' year], 'FontSize', 16, 'FontWeight', 'bold') % Include 'Virtual ' before 2020
```
again...
says "breakout 2" instead of "breakout 3" and comments on 24 are not relevant to breakout 3. The comments relate to the particular bug/requested edit, so they need to be relevant.

alcantarar reviewed (changes requested):
Gary, this PR was a bit premature. Two big picture comments:
1. The markdown handout should walk them through the process of reverting changes, not just giving them a hint. This is a new git concept for them and thus the handout should more closely resemble that of breakout 1 instead of breakout 2. Additionally, each team member should have something to do. Currently, only 3 members (whoever is associated with script 1, 2, and 5) would be doing things while the others just wait around.
2. Please put solutions back in the main script files. It maintains the simplicity of the repository and ultimately, this is not a tutorial on programming. I think this tutorial will be challenging enough for some people without us hiding the solutions, especially when the point of the whole thing is git/github.

GBruening committed:
Fixed some of ryans comments didnt change some others

GBruening commented:
I hate the solutions in the script files.
If you want to put them there I'll let you. But if you don't want them copying and pasting like you said in another comment, then they shouldn't be there. The changes are so simple, if they can't figure it out they won't be able to figure out git.

The markdown had information on checking out to a specific branch, and I'm not sure where it went. I don't think it should be addressed here how to checkout to specific commits but should be done earlier.

This checkout method is going to be a little odd. Because they 'll need to go into their branch and checkout that branch, then pull request again.

GBruening committed:
adding details on how to do gitcheckout

GBruening committed:
Added a few pictures

GBruening committed:
Making commits.png better

GBruening committed:
Add files via upload

GBruening committed:
Delete commits.png

GBruening committed:
Add files via upload

GBruening committed:
Delete commits.png

GBruening committed:
Add files via upload

alcantarar commented:
> I hate the solutions in the script files.

🙄 

> If you want to put them there I'll let you. But if you don't want them copying and pasting like you said in another comment, then they shouldn't be there. The changes are so simple, if they can't figure it out they won't be able to figure out git.

Yep, no solution files. It was simpler before and allows for easy reference because again, the point of the tutorial isn't how to debug or see if they can figure it out. We need this tutorial to be as user-friendly as possible.

> The markdown had information on checking out to a specific branch, and I'm not sure where it went. I don't think it should be addressed here how to checkout to specific commits but should be done earlier.
>
> This checkout method is going to be a little odd. Because they 'll need to go into their branch and checkout that branch, then pull request again.

We can talk tomorrow about what this breakout room is/isn't supposed to be about.

alcantarar committed:
major changes to breakout3 because checking out commits results in merge conflicts

Ideally the simplest way to undo these changes is to checkout a past commit, but getting
each attendee to do this results in merge conflicts. Because we don't want to get into
rebasing, we're having attendees just revert their pull requests. This worksheet has them
doing the following:
1. finding their pull request on github
2. reverting it and merging to master
3. viewing repository network graph to visualize branches
4. read some future considerations about issues/project boards, and an alternative way
to undo changes in a repository.

alcantarar committed:
add comment reminding users to make sure their matlab working dir is their repo folder

alcantarar committed:
rm script_1_sol and comment out solutions for script1

alcantarar committed:
cleanup commented out parts

alcantarar committed:
rm script2 solution and comment out solutions in script2

alcantarar committed:
rm script3 solution and comment out solutions in script3

alcantarar committed:
rm script4 solution and comment out solutions in script4

alcantarar committed:
rm script4 and 5, comment out solutions in script5

alcantarar committed:
name breakout docs consistently

alcantarar committed:
make reviewer 2's suggestsions really bad/ugly.

alcantarar committed:
fix bug in script_5 breakout 2 and make text green

alcantarar committed:
add reference plots for breakout 1 and 2

alcantarar committed:
rm colorblind reference JIC

alcantarar committed:
clarify opening line of breakout 3

alcantarar commented:
OK this PR is ready for review @GBruening. 

- made changes to breakout 3 after our revert/checkout experiments today
- removed solution files (sorry not sorry)
- fixed bug in script 5
- made reviewer 2's suggestions **really** ugly
- confirmed all solutions ran on my machine, which produced the figures below.
- @swkasica, this PR's MATLAB scripts should be ready for porting. Most changes moving forward will be to `breakout_3.md`

post reviewer comments:
![breakout2_fig](https://user-images.githubusercontent.com/38192529/88226457-54674180-cc29-11ea-90b3-0c53f37c6009.png)
finished/original figure:
![breakout1_fig](https://user-images.githubusercontent.com/38192529/88226462-55986e80-cc29-11ea-961e-2bf3cf07b0c8.png)

alcantarar commented:
I guess gary can't review this. I'll merge this because it's at least functional and we can address revisions later.

GBruening commented:
I can review it but do that later probably, and I made this comment in the issues.

If we have the solutions in the file, it encourages them to just copy paste it or uncomment and comment things. Literally 0 investment into what they actually need to do. If they actually have to do something, they'll have a better chance of remembering it because theres more investment. The problems are not hard to fix, if you hand them everything on a plate they'll forget it all
